internal/internal_worker_base.go

```diff
@@ -244,7 +245,10 @@ func (bw *baseWorker) isShutdown() bool {
 
 func (bw *baseWorker) runPoller() {
 	defer bw.shutdownWG.Done()
+	defer bw.options.statsCollector.ShutdownPoller()
+
 	bw.metricsScope.Counter(metrics.PollerStartCounter).Inc(1)
+	bw.options.statsCollector.StartPoller()
```

Comment:
do we need one statsCollector per poller? It doesn't know anything about poller (it seems a worker level component) so not sure what we expect from it per poller

Reply:
It's per worker actually. I am calling these methods here to track the running pollers per worker but I plan to add more methods for collection on running activities and workflows on a worker. So it is on a worker level indeed